### ISSUE TITLE
Makecode extension

### ISSUE THREAD

**Issue by compie67** [ElectronicCats/pxt-airQuality#2]
He, is there a way to obtain the CC811 extension to the microbit Makecode 
can not find it any where like your picture

thank you in advance

**Comment by sabas1080** [comment]
Hi @compie67 , yes only insert or copy&past url  this repo in extension
<img width="1440" alt="Captura de Pantalla 2020-09-17 a la(s) 20 50 51" src="https://user-images.githubusercontent.com/2874292/93546055-95c55600-f927-11ea-8688-0eca05e733ac.png">

and ready
<img width="1436" alt="Captura de Pantalla 2020-09-17 a la(s) 20 52 09" src="https://user-images.githubusercontent.com/2874292/93546095-b1306100-f927-11ea-95f9-2921f9fb6cee.png">

**Comment by compie67** [comment]
first thank you for taking time to answer, i am helping out some students and if i use your provided link i get this
same for the Windows10 app, 
![Schermafbeelding 2020-09-18 074448](https://user-images.githubusercontent.com/6383681/93560808-3e54d380-f983-11ea-9404-a9c82d7c4c19.png)

so weird

**Comment by compie67** [comment]
Update!!!! found it, trange change it to english and there it is, okay now another problem ;-)

I am missing the start block CC811. (CCS811 Start Function
The first block in the AirQuality package is the |CCS811 Start| block. At the start of any program which will use the air quality sensor place the |CCS811 Start| block in a "Forever" block above the rest of your code.)

Great that we are one step further.
regards Eric

**Comment by sabas1080** [comment]
Excellent, the device starts automatically when reading data

**Comment by compie67** [comment]
Hey @sabas1080 
it wont work, like i said we are missing the start CC811 block like in the picture. here
https://github.com/ElectronicCats/pxt-airQuality

or do you have an axample code with your code for us to use.

If got the basic working, we then can let the students try to extract the data to excel etc.

**Comment by sabas1080** [comment]
@compie67 We removed start in the commit https://github.com/ElectronicCats/pxt-airQuality/commit/dafaad52556824c52fee532b1582dd422245a8d8

New example is

```
let eCO2 = 0
let TVOCs = 0
loop.forever(() => {
    TVOCs = airQuality.readTvoc()
    eCO2 = airQuality.readCo2()
    loop.pause(1000)
})
```
I'm going to update the README

**Comment by compie67** [comment]
goodmorning(here in the Netherlands)
our mindset is makecode microbit, this code can not be used in that.

Do we heve to use arduino?

**Comment by sabas1080** [comment]
No, is javascript mode

<img width="1440" alt="Captura de Pantalla 2020-09-19 a la(s) 1 42 17" src="https://user-images.githubusercontent.com/2874292/93660957-9f2eeb00-fa19-11ea-8ef1-2e994827f7da.png">

but is this a example in blocks
<img width="1440" alt="Captura de Pantalla 2020-09-19 a la(s) 1 43 54" src="https://user-images.githubusercontent.com/2874292/93660960-a950e980-fa19-11ea-9bf2-d3a930815e83.png">


Good night in México

**Comment by compie67** [comment]
bam, no were talking.
![image](https://user-images.githubusercontent.com/6383681/93661207-4762a500-fa56-11ea-9881-c47b8d36c8f0.png)
okay, will try to test this today and let you know
eric

**Comment by compie67** [comment]
hi, wel got it working, but it keep giving the value zero. on both "read options"

![IMG_20200922_142812 (1)](https://user-images.githubusercontent.com/6383681/93886998-67a89300-fce6-11ea-98b2-bb7b41621ea3.jpg)
![IMG_20200922_142748](https://user-images.githubusercontent.com/6383681/93887001-68412980-fce6-11ea-8b99-f929bbae1225.jpg)

**Comment by sabas1080** [comment]
Try with this example:

<img width="510" alt="Captura de Pantalla 2020-09-22 a la(s) 10 41 24" src="https://user-images.githubusercontent.com/2874292/93905156-42089300-fcc0-11ea-8b24-89b47074c0ee.png">

I have change Address and Device Status log, send me your "Device Status" if you have error

**Note: Connect PIN WAK to GND**

**Comment by compie67** [comment]
Hi, again also on behave of the students.
Thank you 

ok, i disconnect the CCS811 module from the board and connect this directly on the wire.
Red goes to 3.3v Microbit
green goes to GND microbit

blue goes from p20 microbit to SDA
and purple goes from p19 to SCL

upload the code and status gives me also value zero

**Comment by sabas1080** [comment]
Are you connecting PIN WAK to GND?

**Comment by compie67** [comment]
Do I need this. No I do not use wake

Op wo 23 sep. 2020 19:21 schreef Sabas <notifications@github.com>:

**Comment by compie67** [comment]
goodmorning!! well connect the WAK to gnd and its now on. 07:35 amsterdam time. I give it 20 minute burn time and wait to see what happen. Nect step is to try another CCS811(i bought three) maybe the one i got was broken
but for now all options are giving the result zero

**Comment by sabas1080** [comment]
Good night 0:39 Mexico tine, I am going to test the sensor these days with a microbit and a sensor

**Comment by compie67** [comment]
cool, hope you find something

**Comment by sabas1080** [comment]
Hi @compie67 I have tested and working for me , my configuration is:
CCS811  -  Microbit
GND      --   GND
VCC       --  VCC
SDA       --  SDA
SCL       --  SCL
WAK      --  GND

My code is:
<img width="375" alt="Captura de Pantalla 2020-09-24 a la(s) 15 28 28" src="https://user-images.githubusercontent.com/2874292/94196875-9a839000-fe7a-11ea-83fe-b909a9a1eef9.png">

My video
https://twitter.com/Sabasacustico/status/1309227083814572033

**Comment by compie67** [comment]
wow, and yess got the same now.
and same numbers as yours ;-) and that stay's that way

co2 gives me 4605
tvoc gives me 65021

it wont drop, or go up
but, i got number!!!! 
https://photos.app.goo.gl/YinUWU9hJPEQzyueA

so so far so good.
did you get any other number, like if you blow at the sensor?

**Comment by compie67** [comment]
o, this also work
![image](https://user-images.githubusercontent.com/6383681/94258933-f4915d80-ff2d-11ea-923a-ad854289e452.png)

that is the goal of our project(student need to sort this out there self) ;-)
but as you can see, it keeps giving the same data

**Comment by sabas1080** [comment]
It seems that I forgot to put a reset sequence, can you try the new version 0.1.4 in a new project?

**Comment by sabas1080** [comment]
With new code I have
co2 gives me 3327
tvoc gives me 65021

Also, AMS recommends that you run this sensor for 48 hours when you first receive it to "burn it in", and then 20 minutes in the desired mode every time the sensor is in use. This is because the sensitivity levels of the sensor will change during early use.

**Comment by compie67** [comment]
okay, seems legit. you said new code? 0.1.4 is that also block or java?
cant find it

**Comment by sabas1080** [comment]
is block, only open a new project and import extension

**Comment by compie67** [comment]
Great. Did not know that.
Okay tomorrow morning I wil try it and let it run
Keep you posted.


Op za 26 sep. 2020 21:30 schreef Sabas <notifications@github.com>:

**Comment by compie67** [comment]
hi, goodmorning. maybe missing a step, with new project and import extension nothing new happen. Still the same blocks

**Comment by compie67** [comment]
So, update.

manage to get version 0.1.4
did you get any other reading?
![image](https://user-images.githubusercontent.com/6383681/94476464-ced7b300-01d0-11eb-9488-a8470168d63a.png)

after three days still give the reading co2 4605

**Comment by compie67** [comment]
hey, how are you doing.
status update, weird thing. Fresh code uploaded to microbit. It works and its giving nice value
poweroff and al go back to the weird reading 4605 for co2 and 65021 and stay that way.
and i get it work anymore. 

To bad, looks good to work for the students

**Comment by compie67** [comment]
@sabas1080 got this answer on the static value

quote:
The CCS811 has a resistor Rs that varies with air quality. The tvoc value is roughly Rs/R0, where R0 is the resistance for cleanest air seen by the sensor. R0 is also known as "baseline" and is maintained by the sensor as long as it is powered. Once the sensor is switched off, R0 is lost. After a new power on, it takes some time before a good R0 is determined by the sensor.

So, what you can do is x=get_baseline () before you switch off, and set_baseline(x) after you switch on. But read the datasheet for details on when to do this exactly.
en quote

Is this possible with the MicroBit?

**Comment by Marshall-Jed** [comment]
@compie67 I've had similar issues to you. I ended up finding that this library worked well using an adafruit ccs881 board https://github.com/xinabox/pxt-SG33. Example: https://makecode.microbit.org/_A43RqUhbzfyF

**Comment by redegelde** [comment]
@Marshall-Jed and after powerdown and powerup it give the proper value again?

**Comment by Marshall-Jed** [comment]
Yep, tested several times. Still have the 20 minute baseline issue - I see
you switch to SGP30 which has onboard memory to store this for 7 days.

Will try it with sparkfun and some other sensors too.
